### PR TITLE
Handle cart with virtual products

### PR DESCRIPTION
**Pull request by maximebiloe**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | When your cart contains only virtual products, there is no need to ask for a delivery address, a carrier and also it's useless to display shipping costs. |
| Type? | improvement |
| Category? | FO |
| BC breaks? | no |
| Deprecations? | no |

### CODE REVIEW

File: themes/classic/templates/checkout/_partials/steps/addresses.tpl [hunk 1]
```diff
@@ -42,7 +46,7 @@
           <a href="{$new_address_delivery_url}"><i class="material-icons">&#xE145;</i>{l s='add new address' d='Shop.Theme.Actions'}</a>
```
Comment (kpodemski) [inline themes/classic/templates/checkout/_partials/steps/addresses.tpl]: I guess that this is also not needed?

File: src/Adapter/Cart/CartPresenter.php [hunk 2]
```diff
@@ -273,15 +273,17 @@ public function present($cart)
...
-        $subtotals['shipping'] = array(
-            'type' => 'shipping',
-            'label' => $this->translator->trans('Shipping', array(), 'Shop.Theme.Checkout'),
-            'amount' => $shipping_cost,
-            'value' => $shipping_cost != 0
-                ? $this->priceFormatter->format($shipping_cost)
-                : $this->translator->trans('Free', array(), 'Shop.Theme.Checkout'),
-        );
+        if (!$cart->isVirtualCart()) {
+            $shipping_cost = $cart->getTotalShippingCost(null, $this->includeTaxes());
```
Comment (mickaelandrieu) [inline src/Adapter/Cart/CartPresenter.php]: would you mind to update the variable name to camelcase form ? ``$shippingCost``

File: src/Adapter/Order/OrderPresenter.php [hunk 3]
```diff
@@ -141,13 +141,16 @@ private function getAmounts(Order $order)
...
-        $shipping_cost = ($this->includeTaxes()) ? $order->total_shipping_tax_incl : $order->total_shipping_tax_excl;
-        $subtotals['shipping'] = array(
-            'type' => 'shipping',
-            'label' => $this->translator->trans('Shipping and handling', array(), 'Shop.Theme.Checkout'),
-            'amount' => $shipping_cost,
-            'value' => $shipping_cost != 0 ? $this->priceFormatter->format($shipping_cost) : $this->translator->trans('Free', array(), 'Shop.Theme.Checkout'),
-        );
+        $cart = new Cart($order->id_cart);
+        if (!$cart->isVirtualCart()) {
+            $shipping_cost = ($this->includeTaxes()) ? $order->total_shipping_tax_incl : $order->total_shipping_tax_excl;
```
Comment (mickaelandrieu) [inline src/Adapter/Order/OrderPresenter.php]: same here :)

Comment (maximebiloe) [inline src/Adapter/Order/OrderPresenter.php]: There is only one method, in the cart. Here we get back the value in database.

**Comment by mickaelandrieu** [comment]
Thank you @maximebiloe !

**Pull request merged**